### PR TITLE
Add search_numpy_blas and blas_lib parameters to CompileNativeOpJob

### PR DESCRIPTION
**Pull request by curufinwe**
tk.uncached_path used for blas_lib because the location in my setup is in a singularity image and thus the external sisyphus manager cannot check for it.

### CODE REVIEW

File: returnn/compile.py [hunk 1]
```diff
@@ -116,16 +116,22 @@ class CompileNativeOpJob(Job):
...
         returnn_root=None,
+        search_numpy_blas=True,
+        blas_lib=None,
     ):
         """
         :param str native_op: Name of the native op to compile (e.g. NativeLstm2)
         :param Path|str returnn_python_exe: file path to the executable for running returnn (python binary or .sh)
         :param Path|str returnn_root: file path to the RETURNN repository root folder
+        :param bool search_numpy_blas: search for blas lib in numpy's .libs folder
+        :param Path|str blas_lib: explicit path to the blas library to use
```
Comment (JackTemaki) [inline returnn/compile.py:134]: Can we not establish the convention that paths always have to be `tk.Path`? Then you also do not need the `tk.uncached_path()` below, and for constant stuff you can just wrap it with `Path(...)` in the setup...

**Review by JackTemaki — CHANGES_REQUESTED** [review]

File: returnn/compile.py (hunk 1 above)
Comment (curufinwe) [reply returnn/compile.py:134]: In this case I cannot pass a path from the outside as the blas_lib is only inside a singularity container and not visible from the manager.

Comment (curufinwe) [reply returnn/compile.py:134]: i.e. the manager will complain about missing input, but the Path is there during the runtime of the job.

Comment (JackTemaki) [reply returnn/compile.py:134]: Then it is okay for this case, but maybe we should add a "no check" flag to a Path object then...

**Review by JackTemaki — APPROVED** [review]

File: returnn/compile.py (hunk 1 above)
Comment (curufinwe) [reply returnn/compile.py:134]: Yes, that would be a more principled thing.

**Pull request merged**